File: clojure/gigasecond/src/gigasecond.clj
(ns gigasecond)

(defn from [year month day]  
   (let [date (.plusDays (java.time.LocalDate/of year month day)(/ 1000000000 (* 3600 24)))] 
      [(.getYear date) (.getMonthValue date) (.getDayOfMonth date)]))

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    private LocalDateTime dateTime;
    
    public Gigasecond(LocalDate moment) {
        this.dateTime = moment.atStartOfDay().plusSeconds(1000000000);
    }

    public Gigasecond(LocalDateTime moment) {
        this.dateTime = moment.plusSeconds(1000000000);
    }

    public LocalDateTime getDateTime() {
        return this.dateTime;
    }
}
